apps/api/src/modules/job/job.repository.ts

```diff
@@ -153,6 +153,74 @@ export class JobRepository {
     return selectedFields
   }
 
+  async getJobApplied(
+    titleSearch: string,
+    statusQuery: JobStatus[],
+    applicationStatus: ApplicationStatus[],
+    userId: number,
+  ): Promise<(GetJobCardDto & { ApplicationStatus })[]> {
+    const jobs = await this.prisma.job.findMany({
+      orderBy: {
+        jobId: Prisma.SortOrder.desc,
+      },
+      include: {
+        Casting: {
+          include: {
+            User: true,
+          },
+        },
+        //only get the latest application
+        Application: {
+          orderBy: {
+            applicationId: Prisma.SortOrder.desc,
+          },
+          take: 1,
+        },
```

Comment:
This will take latest application not the actor's application
```suggestion
        Application: {
          where: {
            actorId: userId,
          },
        },
```